molgenis-one-click-importer/src/main/frontend/src/store/actions.js

```diff
@@ -1,42 +1,25 @@
-import fetch from 'isomorphic-fetch'
+import api from '@molgenis/molgenis-api-client'
 import { ADD_FINISHED_JOB, UPDATE_JOB } from './mutations'
 
 export const IMPORT_FILE = '__IMPORT_FILE__'
 export const POLL_JOB_URL = '__POLL_JOB_URL__'
 
 export default {
   [IMPORT_FILE] ({dispatch}, file) {
-    const formData = new FormData()
-    formData.append('file', file)
-
-    const options = {
-      body: formData,
-      method: 'POST',
-      credentials: 'same-origin'
-    }
-
-    fetch('/plugin/one-click-importer/upload', options).then(response => {
-      response.text().then(jobUrl => {
-        dispatch(POLL_JOB_URL, jobUrl)
-      })
+    api.postFile('/plugin/one-click-importer/upload', file).then(response => {
+      dispatch(POLL_JOB_URL, response)
     })
   },
   [POLL_JOB_URL] ({dispatch, commit}, jobUrl) {
```

Comment:
The function poll_job_url is not called from outside the actions, and therefore should be private and then tested as part of import_file.

Reply:
I tried moving it, but due to the polling mechanism inside the POLL_JOB_URL action, it won't work because the Promise never gets closed.

You could say that the second time the api.get(...) is called you want to return a job object which is Failed or Finished. But I haven't figured out how to to this with testdouble.